script.js

```diff
@@ -1,4 +1,159 @@
+};
+
+// function for player to roll 2 dice to get 2 number
+var chooseDiceOrder = function (userInput) {
+  orderOfDice = userInput;
+  //input validation > error message for player if they did not enter 1 or 2
+  if (userInput != 1 && userInput != 2) {
+    return "Please enter either 1 or 2 to choose which dice to put infront.";
+  }
+
+  //if user choose 1 then first dice will be infront and second dice will be behind.
+
+  if (userInput == "1") {
+    var diceOrder = Number(String(playerRoll1) + String(playerRoll2));
+    console.log(`user chose ${playerRoll1}, then ${playerRoll2}.`);
+    //game mode switches to the next player
+    gameMode = "next player roll 2 dice";
+    console.log(gameMode);
+    myOutputValue = `Player ${currentPlayer}, you have chosen dice 1. <br><br>Your final value is ${playerRoll1}${playerRoll2}.<br><br> Please click submit.`;
+  }
+
+  //if user choose 1 then first dice will be infront and second dice will be behind.
+  if (userInput == "2") {
+    diceOrder = Number(String(playerRoll2) + String(playerRoll1));
+    console.log(`user chose ${playerRoll2}, then ${playerRoll1}.`);
+    gameMode = "next player roll 2 dice";
+    //game mode switches to the next player
+    console.log(gameMode);
+    myOutputValue = `You have chosen 2. <br><br>Your final value is ${playerRoll2}${playerRoll1}. <br> <br> Please click submit.`;
+  }
+
+  //to push the user's chosen number into an array and empty the temporary variables
+
+  if (accumulativeRound == 0) {
+    allDiceOrder.push(diceOrder);
+    console.log(allDiceOrder);
+  }
+
+  if (accumulativeRound >= 1) {
+    if (currentPlayer == 1) {
+      allDiceOrder[0] = allDiceOrder[0] + diceOrder;
+      console.log(`new array value = ${allDiceOrder[0]}`);
+      diceOrder = [];
+    }
+    if (currentPlayer == 2) {
+      allDiceOrder[1] = allDiceOrder[1] + diceOrder;
+      console.log(`new array value = ${allDiceOrder[0]}`);
+      diceOrder = [];
+    }
+    playerRoll1 = [];
+    playerRoll2 = [];
+  }
+  return myOutputValue;
+};
+
+//functiont to determine the winner by comaring the values in the array
+var determineWinner = function () {
+  var myOutputValue = `Player 1 score is ${allDiceOrder[0]}<br>
+    Player 2 score is ${allDiceOrder[1]}<br>`;
+  var endOfSentence = `This is not the end! Click submit again to continue the game and try to beat your friend!  `;
+  if (allDiceOrder[0] >= allDiceOrder[1]) {
+    myOutputValue =
+      myOutputValue +
+      `<br><br><b>The winner is Player 1.</b><br><br>` +
+      endOfSentence;
+  }
+  if (allDiceOrder[1] >= allDiceOrder[0]) {
+    myOutputValue =
+      myOutputValue +
+      `<br><br> <b>The winner is Player 2!</b><br><br>` +
+      endOfSentence;
+  }
+  if (allDiceOrder[1] == allDiceOrder[0]) {
+    myOutputValue =
+      myOutputValue + `<br><br> <b>It is a draw!</b><br><br>` + endOfSentence;
+  }
+
   return myOutputValue;
 };
```

Comment:
this is good but should line 95 and 101 be a strictly more than condition? since in line 107 we are accounting for ties